python/tests/mode_coeffs.py

```diff
@@ -36,27 +34,44 @@ def run_mode_coeffs(self, mode_num, kpoint_func):
 
         # mode frequency
         fcen = 0.20  # > 0.5/sqrt(11) to have at least 2 modes
+        df   = 0.5*fcen
 
-        sources = [mp.EigenModeSource(src=mp.GaussianSource(fcen, fwidth=0.5*fcen),
-                                      eig_band=mode_num,
-                                      size=mp.Vector3(0,sy-2*dpml,0),
-                                      center=mp.Vector3(-0.5*sx+dpml,0,0),
-                                      eig_match_freq=True,
-                                      eig_resolution=32) ]
+        source=mp.EigenModeSource(src=mp.GaussianSource(fcen, fwidth=df),
+                                  eig_band=mode_num,
+                                  size=mp.Vector3(0,sy-2*dpml,0),
+                                  center=mp.Vector3(-0.5*sx+dpml,0,0),
+                                  eig_match_freq=True,
+                                  eig_resolution=2*resolution)
 
         sim = mp.Simulation(resolution=resolution,
                             cell_size=cell_size,
                             boundary_layers=boundary_layers,
                             geometry=geometry,
-                            sources=sources,
+                            sources=[source],
                             symmetries=[mp.Mirror(mp.Y, phase=1 if mode_num % 2 == 1 else -1)])
 
         xm = 0.5*sx - dpml  # x-coordinate of monitor
-        mflux = sim.add_mode_monitor(fcen, 0, 1, mp.ModeRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
-        mode_flux = sim.add_flux(fcen, 0, 1, mp.FluxRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
+        mflux = sim.add_mode_monitor(fcen, df, nf, mp.ModeRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
+        mode_flux = sim.add_flux(fcen, df, nf, mp.FluxRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
 
         # sim.run(until_after_sources=mp.stop_when_fields_decayed(50, mp.Ez, mp.Vector3(-0.5*sx+dpml,0), 1e-10))
-        sim.run(until_after_sources=100)
+        sim.run(until_after_sources=200)
+
+        ##################################################
+        # If the number of analysis frequencies is >1, we
+        # are testing the unit-power normalization
+        # of the eigenmode source: we observe the total
+        # power flux through the mode_flux monitor (which
+        # equals the total power emitted by the source as
+        # there is no scattering in this ideal waveguide)
+        # and check that it agrees with the prediction
+        # of the eig_power() class method in EigenmodeSource.
+        ##################################################
+        if nf>1:
+            power_observed=mp.get_fluxes(mode_flux)
+            freqs=[mode_flux.freq_min + n*mode_flux.dfreq for n in range(len(power_observed))]
```

Comment:
Instead of computing `freqs` manually, you can use the built-in routine [`get_flux_freqs`](https://meep.readthedocs.io/en/latest/Python_User_Interface/#flux-spectra):
```py
freqs=mp.get_flux_freqs(mode_flux)
```

Reply:
I switched to this, and it now seems clear that the observed DFT data need to be plotted with a leftward shift of 0.001 (i.e. one-half the spacing between X tics) to fall correctly on top of the predicted data. In particular, the maximum entry in the 51-element `flux()` array computed for the DFT cell is not the entry in cell #26, i.e. the exact middle of the array, but rather for cell #27, corresponding to a frequency slightly beyond the center frequency. I don't know what that is about.

Reply:
This is because [`get_flux_freqs`](https://github.com/NanoComp/meep/blob/master/python/simulation.py#L2577-L2578) is based on Numpy's [`linspace`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.linspace.html) and is using `endpoint=False`:
```
def get_flux_freqs(f):
    return np.linspace(f.freq_min, f.freq_min + f.dfreq * f.Nfreq, num=f.Nfreq, endpoint=False).tolist()
```
Thus for an *odd* number of frequency bins, the center frequency will *not* be `fcen` and will *not* be the middle element of the `freqs` array (e.g., the 26th element of the 51-element array). Rather, the center frequency will be the element adjacent to the middle element (e.g., the 27th element of the 51-element array). For reference, this is demonstrated below.
```
>>> import numpy as np
>>> fcen = 0.20; df = 0.5*fcen; nfreq = 51;
>>> fmax = fcen+0.5*df; fmin = fcen-0.5*df;
>>> dfreq = (fmax-fmin)/(nfreq-1);
>>> freqs = np.linspace(fmin, fmin+dfreq*nfreq, num=nfreq, endpoint=False);
>>> print("26th element: {}".format(freqs[25]))
26th element: 0.2
>>> print("27th element: {}".format(freqs[26]))
27th element: 0.202
```
The size of the frequency spacing (i.e. `dfreq` from above) is defined in [`fields::add_dft`](https://github.com/NanoComp/meep/blob/master/src/dft.cpp#L166).